mcs/class/System.Private.CoreLib/System.IO/Stream.cs

```diff
@@ -2,14 +2,8 @@ namespace System.IO
 {
 	partial class Stream
 	{
-		private bool HasOverriddenBeginEndRead ()
-		{
-			throw new NotImplementedException ();
-		}
+		bool HasOverriddenBeginEndRead () => true;
```

Comment:
This is incorrect, please implement it correctly instead

Reply:
Well, it's correct, it's just won't use an optimization https://github.com/dotnet/coreclr/blob/d0cdf8813ef099deec3b6940e34f20d1bba054b8/src/System.Private.CoreLib/shared/System/IO/Stream.cs#L1255-L1260 (CoreRT ignores it too).

I can try to implement an icall to scan method table for overriden Begin/End Write/Read.

Reply:
I'll merge this but please implement this so we can benefit from the optimization